src/librustc_resolve/late.rs

```diff
@@ -2160,8 +2160,10 @@ impl<'a, 'b, 'ast> LateResolutionVisitor<'a, 'b, 'ast> {
                     self.resolve_expr(argument, None);
                 }
             }
-            ExprKind::Type(ref type_expr, _) => {
-                self.diagnostic_metadata.current_type_ascription.push(type_expr.span);
+            ExprKind::Type(ref type_expr, ref ty) => {
+                self.diagnostic_metadata
+                    .current_type_ascription
+                    .push(type_expr.span.between(ty.span));
```

Comment:
The setup here is very subtle and needs a comment.
`ParseSess::type_ascription_path_suggestions` keeps spans of colon tokens in type ascription.
Here we are trying to retrieve the span of the colon token as well, but only if it's written without spaces `expr:Ty` and therefore confusable with `expr::Ty`, only in this case it will match the span from `type_ascription_path_suggestions`.